pkg/store/bucket.go

```diff
@@ -190,6 +191,10 @@ func newBucketStoreMetrics(reg prometheus.Registerer) *bucketStoreMetrics {
 		Name: "thanos_bucket_store_queries_dropped_total",
 		Help: "Number of queries that were dropped due to the sample limit.",
 	})
+	m.queriesSeriesDropped = promauto.With(reg).NewCounter(prometheus.CounterOpts{
+		Name: "thanos_bucket_store_queries_series_dropped_total",
+		Help: "Number of queries that were dropped due to the series limit.",
```

Comment:
As I can see that if we want to add more metrics for the limit, why we just reuse `thanos_bucket_store_queries_dropped_total` metric? We can add a label `reason` for the reason of query drop such as `series`, `samples`, `chunks`, etc.

Reply:
If modify `thanos_bucket_store_queries_dropped_total`, it will affect `ChunksLimiterFactory`. Cortex depending it.

Reply:
No, the factory interface `ChunksLimiterFactory` won't be changed. The only change would be making `thanos_bucket_store_queries_dropped_total` a `CounterVec` type and adding one more label to it.

When initializing chunksLimiter you just need to do something like below so that's the same.

```
chunksLimiter    = s.chunksLimiterFactory(s.metrics.queriesDropped.WithLabelValues("chunks"))
```

It is also good to keep the current one because chunks and series are two dimensions. We can change it later if needed.

Reply:
Yes, you are right

Reply:
Nice! Is it still todo?

Reply:
I like this idea (:

Reply:
Okay, let me modify it.